src/solvers/flattening/boolbv_quantifier.cpp

```diff
@@ -140,11 +138,14 @@ exprt get_quantifier_var_max(
 bool instantiate_quantifier(exprt &expr,
                             const namespacet &ns)
 {
-  if(!(expr.id()==ID_forall || expr.id()==ID_exists))
-    return true;
+  PRECONDITION(expr.id() == ID_forall || expr.id() == ID_exists);
 
-  assert(expr.operands().size()==2);
-  assert(expr.op0().id()==ID_symbol);
+  DATA_INVARIANT(
+    expr.operands().size() == 2,
+    "quantifier expressions shall have two operands");
```

Comment:
I believe we have `to_quantifier_expr` by now, and that method should actually take a `quantifier_exprt`.